test/Test/Pos/Ssc/GodTossing/ComputeSharesSpec.hs

```diff
@@ -44,13 +60,184 @@ spec = describe "computeSharesDistr" $ do
+    | otherwise = do
+        let n = length rs
+        let totalCases = toRational (2::Int)^n :: Rational
+        let fivePerc = 0.05 :: Rational
+        let totalCoins = sum $ map unsafeGetCoin $ toList rs
+        let totalDistr = sum (toList sd)
+        let stakeholders = HM.keys rs
+        let distrs = map (findStk totalCoins totalDistr) stakeholders
+        let !(er, firstCase, secondCase) = fairBrute distrs 0 0
+        er < 0.05 &&
+            toRational firstCase < totalCases * fivePerc &&
+            toRational secondCase < totalCases * fivePerc
+  where
+    fairBrute :: [(Rational, Rational)] -> Rational -> Rational -> Unfairness
+    fairBrute [] real generated = computeUnfairness real generated
+    fairBrute ((r, g):xs) !real !generated
+        -- We get the situation when both these values more than 0.5,
+        -- so we can't get some unfairness afterwards.
+        | real > 0.5 && generated > 0.5 = (0, 0, 0)
+        | otherwise = do
+            let !whenNodeOffline = fairBrute xs real generated
+            let !whenNodeOnline  = fairBrute xs (real + r) (generated + g)
+            whenNodeOnline `combineUnfair` whenNodeOffline
+    divRat a b = toRational a / toRational b
+    findStk :: Word64 -> Word16 -> StakeholderId -> (Rational, Rational)
+    findStk totalCoins totalDistr stId = do
+        let rFrac = fromMaybe (error "Real stake isn't found")
+                              (unsafeGetCoin <$> HM.lookup stId rs) `divRat` totalCoins
+        let gFrac = fromMaybe (error "Distribution isn't found")
+                              (HM.lookup stId sd) `divRat` totalDistr
+        (rFrac, gFrac)
+
+    combineUnfair (er1, c1, c2) (er2, c3, c4) =
+        (max er1 er2, c1 + c3, c2 + c4)
+    computeUnfairness :: Rational -> Rational -> Unfairness
+    computeUnfairness real generated
+        -- Strictly say constant 0.5 depends on number of shares required
+        -- to reaval of commitment.
+        -- Bad case of the first type when
+        -- nodes can reveal commitment using shares
+        -- but real coin distribution says that nodes can't do it.
+        | real < 0.5 && generated > 0.5 = (generated - 0.5, 1, 0)
+        -- Bad case of the second type when
+        -- nodes can't reveal commitment using shares of honest nodes
+        -- but real coin distribution says that nodes can do it.
+        | real > 0.5 && generated < 0.5 = (0.5 - generated, 0, 1)
+        | otherwise = (0, 0, 0)
+
+isDistrReasonable :: Word16 -> RichmenStakes -> Either TossVerFailure SharesDistribution -> Bool
+isDistrReasonable _ _ (Left _) = False
+isDistrReasonable mx rs (Right sd) =
+    -- (1) Sum of distribution is less than some reasonable number.
+    isLimitedBy mx sd &&
+    -- (2) Distribution is fair (see explanation of @isDistrFair@)
+    isDistrFair rs sd
+
+isDistrReasonableMax :: RichmenStakes -> Either TossVerFailure SharesDistribution -> Bool
+isDistrReasonableMax = isDistrReasonable 300
```

Comment:
Shouldn't it be `3 / genesisMpcThd` instead (with appropriate conversions and rounding)?